h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_add_te_model_to_a_regular_model.py

```diff
@@ -0,0 +1,82 @@
+    print(te)
+    print(trainingFrame)
+
+    myX = ["pclass", "sex", "age", "sibsp", "parch", "fare", "cabin", "embarked", "home.dest"]
+    titanic_gbm_model = H2OGradientBoostingEstimator(ntrees=50,
+                                             learn_rate=0.1,
+                                             score_tree_interval=10,
+                                             stopping_rounds=5,
+                                             stopping_metric="AUC",
+                                             stopping_tolerance=0.001,
+                                             distribution="multinomial",
+                                             categorical_encoding="enum_limited",
+                                             te_model_key = te.key,
+                                             seed=1234)
+
+    assert titanic_gbm_model.te_model_key is not None
+
+    titanic_gbm_model.train(x=myX, y=targetColumnName,
+                    training_frame=trainingFrame)
+
+    variable_importance = titanic_gbm_model._model_json['output']['variable_importances'].as_data_frame()
+
+    # Checking that model used te encoded columns during trining
+    var_imp_for_te_encoded_columns = variable_importance[variable_importance['variable'].str.contains('_te', regex=True)]
+    print(var_imp_for_te_encoded_columns)
+    assert len(var_imp_for_te_encoded_columns) == 2
+
+    # Checking that predictions are different when we attach te model
+    preds = titanic_gbm_model.predict(trainingFrame)
+    print(preds)
+
+    # Let' train basically the same Model but without te_model_key
+    titanic_gbm_model_without_te = H2OGradientBoostingEstimator(ntrees=50,
+                                                     learn_rate=0.1,
+                                                     score_tree_interval=10,
+                                                     stopping_rounds=5,
+                                                     stopping_metric="AUC",
+                                                     stopping_tolerance=0.001,
+                                                     distribution="multinomial",
+                                                     categorical_encoding="enum_limited",
+                                                     seed=1234)
+
+    assert titanic_gbm_model_without_te.te_model_key is None
+
+    titanic_gbm_model_without_te.train(x=myX, y=targetColumnName,
+                            training_frame=trainingFrame)
+
+    preds_without_te = titanic_gbm_model_without_te.predict(trainingFrame)
+    print(preds_without_te)
+
+    assert not pyunit_utils.compare_frames_local(preds, preds_without_te, 1, tol=1e-10, returnResult=True), "Predictions should be different"
+
+testList = [
+    test_target_encoder_attached_as_preprocessor
+]
+
+if __name__ == "__main__":
+    for test in testList: pyunit_utils.standalone_test(test)
+else:
+    for test in testList: test()
```

Comment:
fyi, please avoid this in tests, we had an issue with this pattern during last release: when running tests as script of from IDE, it's using the first ( __name__ == "main"), but from our test runner (in Jenkins), it was using the second....
I fixed the test runner since, but this pattern is still nasty.

I recommend to use only following instead:
```
pyunit_utils.run_tests(testList)
```

Reply:
@sebhrusen thanks for info, good to know. Noticed that IDE will not recognise starting point ( no green triangle on the side) with new approach - fortunately with right-click context menu it is still working.

Reply:
yes, 'run' and 'debug' shortcuts also working afaik